### DISCUSSION TITLE
Python and Rust examples

### DISCUSSION THREAD

**Discussion by SonnyWalkman** [microsoft/auteur#10] [Ideas]
Just a thought?? Could a few more example python scripts be added. An Example I have is a AV switcher to show how to switch between multiple sources (say 4-8 inputs?) which can be either cut switched or transition switched between sources to one or multiple destinations like RTMP. Similar to a vision switcher with audio follower would in a TV studio?

I guess too a rust example/s (add a examples folder)to show how to interact with auteur server in pure rust.. 

Couple of more dumb questions..
1. Can auteur have multiple instances running?
2. Could rust control auteur without using controller.rs to avoid json scripting to port 8080? Standalone app with auteur as the core?

I'm just thinking of ideas.

**Comment by MathieuDuponchelle** [comment]
Hey, so regarding example scripts I would rather keep them short and to the point (ie demonstrating a single feature as simply as possible). The reason is that the API isn't formalized yet, so the more scripts there are the more code I have to update when making changes :)

If you think your example qualifies I'll be happy to review it, otherwise if it is a more significant piece of code you could also consider putting it up as its own project, which I can link to in the README :)

Note that that python wrapper is really only an artefact of me wanting to have a simple language to write example scripts in, while still making use of the controller executable in doing so.

If I were to write a more serious standalone python controller, I would not call onto the executable in a subprocess, but rather communicate directly with the auteur instance (using the websockets module).

Rust examples would be nice, however I have no plan of exposing a rust API in auteur, that is more maintenance burden than I'm willing to take on at the moment :)

Auteur can have multiple instances running of course, just spawn them with a different port. Auteur is however not designed for eg load balancing, each instance will be opaque to the others (ie you can't try to connect a source from one instance to a destination in another instance and expect it to work :))

**Reply by SonnyWalkman** [reply]
Thanks for the information, Understand all your points.
